src/ledger/LedgerManagerImpl.cpp

```diff
@@ -1064,6 +1089,27 @@ LedgerManagerImpl::advanceLedgerPointers(LedgerHeader const& header,
     mLastClosedLedger.header = header;
 }
 
+void
+LedgerManagerImpl::maybeUpdateNetworkConfig(bool upgradeHappened,
```

Comment:
I think this code should be behind `
#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION`

Reply:
Since the code compiles fine, I'd prefer to leave this as is. This doesn't touch any XDR and is conditioned on the protocol version at runtime.

Reply:
Actually, I've realized that I've removed the guard from soroban protocol version in this PR as well, so I returned the guard for the sake of consistency and added guard to this function (now if you access settings in vcurr you're guarantteed to crash on accessing nullopt, which is good for catching bugs)